### PR TITLE
Support the 'current' repo

### PR DESCRIPTION
**Pull request by wardhus**
This PR add support for the current directory (which already has a EL_9 build)

### CODE REVIEW

**Comment by wardhus** [comment]
Also fixed the checks and included #118 in this commit.

File: manifests/repository.pp [hunk 1]
```diff
@@ -74,7 +77,12 @@
           $location = "${url}RHEL_${osmajrelease}"
         }
         'Centos', 'Rocky', 'AlmaLinux': {
-          if versioncmp($release, '21') >= 0 {
+          if (
+            (
+              (versioncmp($release, '21') >= 0) or
+              ($release == 'current')
+            ) and (versioncmp($osmajrelease, '8') >= 0)
```
Comment (jhoblitt) [inline manifests/repository.pp:84]: Could you add some unit test coverage of this?

**Comment by jhoblitt** [comment]
The puppet lint errors were fixed by #122. Could you rebase on current master?

**Comment by matejzero** [comment]
I think this needs an extra fix as the URI for repo changed as well. For current, `download.bareos.com/bareos/release/${release}/` is not OK as the repository moved to `download.bareos.org/current/'`

Maybe something like this would work:

```puppet
  if $release == 'current' {
    $uri_path = 'current'
  } else {
    $uri_path = "bareos/release/${release}"
  }

  if $subscription {
    if empty($username) or empty($password) {
      fail('For Bareos subscription repos both username and password are required.')
    }
    # note the .com
    $address = "download.bareos.com/${uri_path}/"
  } else {
    $address = "download.bareos.org/${uri_path}/"
  }
 ```

**Pull request closed without merging**